AzureMonitorAgent/ama_tst/modules/helpers.py

```
@@ -93,16 +111,77 @@ def find_package_manager():
+    try:
+        dpkg_info = subprocess.check_output(['dpkg', '-s', pkg], universal_newlines=True,\
+                                            stderr=subprocess.STDOUT)
+        dpkg_lines = dpkg_info.split('\n')
+        for line in dpkg_lines:
+            if (line.startswith('Package: ') and not line.endswith(pkg)):
+                # wrong package
+                return None
+            if (line.startswith('Status: ') and not line.endswith('installed')):
+                # not properly installed
+                return None
+            if (line.startswith('Version: ')):
+                version = (line.split())[-1]
+                general_info['{0}_VERSION'.format(pkg.upper())] = version
+                return version
+        return None
+    except subprocess.CalledProcessError:
+        return None
+
+def get_rpm_pkg_version(pkg):
+    try:
+        rpm_info = subprocess.check_output(['rpm', '-qi', pkg], universal_newlines=True,\
+                                            stderr=subprocess.STDOUT)
+        if ("package {0} is not installed".format(pkg) in rpm_info):
+            # didn't find package
+            return None
+        rpm_lines = rpm_info.split('\n')
+        for line in rpm_lines:
+            # parse line
+            # note: parsing is weird bc rpm puts various information into two columns
+            parsed_line = line.split()
+            if (parsed_line[0] == 'Name'):
+                # ['Name', ':', name, 'Relocations', ':', relocations]
+                name = parsed_line[2]
+                if (name != pkg):
+                    # wrong package
+                    return None
+            if (parsed_line[0] == 'Version'):
+                # ['Version', ':', version, 'Vendor', ':', 'MSFT']
+                version = parsed_line[2]
+                general_info['{0}_VERSION'.format(pkg.upper())] = version
+                return version
+        return None
+    except subprocess.CalledProcessError:
```

general comment, silently eating errors might not be the appropriate approach
while i understand not wanting to show the user too many technical details that may confuse them, if we don't have any exception output, if there are issues running the TST there is nothing the customer can share with us to help debug (and reproing the issue may be hard for us). it might be better to log these like `Internal exception while <description of function purpose>: <insert exception details>.`